src/components/ShareMyLocationButton.js

```diff
@@ -26,22 +26,39 @@ const ShareMyLocationButton = ({ disabled = false }) => {
   };
   const handleError = () => setUserCenterMap(defaultCenterMap);
 
+  const buttonText = userLocationShared
+    ? 'Stop Sharing Location'
+    : 'Share My Location';
+
   const handleShareLiveLocation = () => {
     if (navigator.geolocation) {
-      if (trackingId === null) {
-        const newTrackingId = navigator.geolocation.watchPosition(
-          handleSuccess,
-          handleError,
-        );
-        setTrackingId(newTrackingId);
-      }
+      const trackingId = navigator.geolocation.watchPosition(
+        handleSuccess,
+        handleError,
+      );
+
+      setTrackingId(trackingId);
     }
   };
 
+  const handleStopSharing = () => {
+    if (userLocationShared && trackingId) {
+      setTrackingId(navigator.geolocation.clearWatch(trackingId));
+    }
+    setUserLocationShared(false);
+    setNewCenterMap(defaultCenterMap);
+  };
+  console.log('---', trackingId);
   return (
-    <button onClick={handleShareLiveLocation} disabled={disabled}>
-      Share My Location
-    </button>
+    <div>
```

Comment:
I think you can get rid of this wrapping `div`. There was a `React.Fragment` there before, but now it's not necessary since you only have one top level element (the `button`).

Reply:
Thank you @laurenmbeatty, it's fixed now